lib/vector-buffers/src/variants/disk_v2/reader.rs

```diff
@@ -610,18 +635,7 @@ where
 
             // If any events were skipped, do our logging/metrics for that.
             if events_skipped > 0 {
-                error!(
-                    dropped_events = events_skipped,
-                    "Detected missing/dropped events.  Buffer data loss has occurred."
-                );
-
-                // TODO: We probably need to make this actually decrement the buffer events gauge directly.
-                //
-                // We don't update it unless there's a received/sent event emitted, which these
-                // events naturally will not be part of as they don't flow out of the buffer.
-                emit(EventsCorrupted {
-                    count: events_skipped,
-                });
+                self.ledger.track_dropped_events(events_skipped);
```

Comment:
Technically, the removed comment still semi-applies, but we've documented it in `track_dropped_events`.